src/tls13.c

```diff
@@ -9720,11 +9730,31 @@ static int SendTls13NewSessionTicket(WOLFSSL* ssl)
     output[idx++] = ssl->session->ticketNonce.data[0];
 
     /* length */
-    c16toa(ssl->session->ticketLen, output + idx);
+    if (IsAtLeastTLSv1_3(ssl->version) &&
+            (ssl->options.mask & WOLFSSL_OP_NO_TICKET) != 0) {
+
+        c16toa(ID_LEN, output + idx);
+    }
+    else {
+        c16toa(ssl->session->ticketLen, output + idx);
+    }
+
     idx += LENGTH_SZ;
     /* ticket */
-    XMEMCPY(output + idx, ssl->session->ticket, ssl->session->ticketLen);
-    idx += ssl->session->ticketLen;
+    if (IsAtLeastTLSv1_3(ssl->version) &&
+            (ssl->options.mask & WOLFSSL_OP_NO_TICKET) != 0) {
+
+        if (ssl->session->haveAltSessionID == 1)
+            XMEMCPY(output + idx, ssl->session->altSessionID, ID_LEN);
+        else
+            XMEMCPY(output + idx, ssl->session->sessionID, ID_LEN);
```

Comment:
We can NOT use the session ID for the ID. It is user provided and can be anything (for example a string of `0`'s).